fastly/custom_tls_activation.go

```diff
@@ -0,0 +1,198 @@
+			"Accept": "application/vnd.api+json", // this is required otherwise the filters don't work
+		},
+	}
+
+	r, err := c.Get(p, filters)
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := jsonapi.UnmarshalManyPayload(r.Body, reflect.TypeOf(new(TLSActivation)))
+	if err != nil {
+		return nil, err
+	}
+
+	a := make([]*TLSActivation, len(data))
+	for i := range data {
+		typed, ok := data[i].(*TLSActivation)
+		if !ok {
+			return nil, fmt.Errorf("unexpected response type: %T", data[i])
+		}
+		a[i] = typed
+	}
+
+	return a, nil
+}
+
+// GetTLSActivationInput is used as input to the GetTLSActivation function.
+type GetTLSActivationInput struct {
+	ID      string
+	Include *string // Include related objects. Optional, comma-separated values. Permitted values: tls_certificate, tls_configuration, and tls_domain.
+}
+
+// GetTLSActivation retrieve a single activation.
+func (c *Client) GetTLSActivation(i *GetTLSActivationInput) (*TLSActivation, error) {
+	if i.ID == "" {
+		return nil, ErrMissingID
+	}
+
+	p := fmt.Sprintf("/tls/activations/%s", i.ID)
+
+	ro := &RequestOptions{
+		Headers: map[string]string{
+			"Accept": "application/vnd.api+json", // this is required otherwise the params don't work
+		},
+	}
+
+	if i.Include != nil {
+		ro.Params = map[string]string{"include": *i.Include}
+	}
+
+	r, err := c.Get(p, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var a TLSActivation
+	if err := jsonapi.UnmarshalPayload(r.Body, &a); err != nil {
+		return nil, err
+	}
+
+	return &a, nil
+}
+
+// CreateTLSActivationInput is used as input to the CreateTLSActivation function.
+type CreateTLSActivationInput struct {
+	ID            string                `jsonapi:"primary,tls_activation"`   // ID value does not need to be set.
+	Certificate   *CustomTLSCertificate `jsonapi:"relation,tls_certificate"` // Only ID of CustomTLSCertificate needs to be set.
+	Configuration *TLSConfiguration     `jsonapi:"relation,tls_configuration"`
+	Domain        *TLSDomain            `jsonapi:"relation,tls_domain"`
+}
+
+// CreateTLSActivation enable TLS for a domain using a custom certificate.
+func (c *Client) CreateTLSActivation(i *CreateTLSActivationInput) (*TLSActivation, error) {
+	if i.Certificate == nil {
+		return nil, ErrMissingTLSCertificate
+	}
+	if i.Configuration == nil {
+		return nil, ErrMissingTLSConfiguration
+	}
+	if i.Domain == nil {
+		return nil, ErrMissingTLSDomain
+	}
+
+	p := "/tls/activations"
+
+	r, err := c.PostJSONAPI(p, i, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var a TLSActivation
+	if err := jsonapi.UnmarshalPayload(r.Body, &a); err != nil {
+		return nil, err
+	}
+
+	return &a, nil
+}
+
+// UpdateTLSActivationInput is used as input to the UpdateTLSActivation function.
+type UpdateTLSActivationInput struct {
+	ID          string                `jsonapi:"primary,tls_activation"`
+	Certificate *CustomTLSCertificate `jsonapi:"relation,tls_certificate,tls_certificate"` // Only ID of CustomTLSCertificate needs to be set.
```

Comment:
Repeat of `tls_certificate`.

```suggestion
	Certificate *CustomTLSCertificate `jsonapi:"relation,tls_certificate"` // Only ID of CustomTLSCertificate needs to be set.
```